### PR TITLE
Custom messages unit tests

### PR DESCRIPTION
**Pull request by maurolacy**
Messages unit tests.

TODO:

- ~~Mint unit tests~~. (done)
- ~~Multi-hop unit tests~~. (done in #16)

### CODE REVIEW

File: app/wasm/message_plugin.go [hunk 1]
```diff
@@ -17,23 +17,23 @@ import (
...
+		return &CustomMessenger{
 			wrapped:    old,
 			bank:       bank,
 			gammKeeper: gammKeeper,
 		}
 	}
 }
 
-type MintTokenMessenger struct {
+type CustomMessenger struct {
```
Comment (ethanfrey) [inline app/wasm/message_plugin.go:29]: Good rename

File: app/wasm/test/messages_test.go [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+	"github.com/osmosis-labs/osmosis/v7/app/wasm"
+	wasmbindings "github.com/osmosis-labs/osmosis/v7/app/wasm/bindings"
+	"github.com/stretchr/testify/assert"
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSwap(t *testing.T) {
```
Comment (ethanfrey) [inline app/wasm/test/messages_test.go:106]: This is cool test that doesn't involve calling the contract, but just the message handler function. Good one

Comment (ethanfrey) [reply app/wasm/test/messages_test.go:106]: Do you want to move some of the multi-hop tests over here?
(Or maybe copy them and keep a few full stack going through the contract)

File: app/wasm/message_plugin.go [hunk 3]
```diff
@@ -71,17 +71,22 @@ func (m *MintTokenMessenger) mintTokens(ctx sdk.Context, contractAddr sdk.AccAdd
 	if err != nil {
 		return nil, nil, sdkerrors.Wrap(err, "sending newly minted coins from message")
 	}
-
 	return nil, nil, nil
 }
 
-func (m *MintTokenMessenger) swapTokens(ctx sdk.Context, contractAddr sdk.AccAddress, swap *wasmbindings.SwapMsg) ([]sdk.Event, [][]byte, error) {
-	_, err := performSwap(m.gammKeeper, ctx, contractAddr, swap)
-	return nil, nil, sdkerrors.Wrap(err, "gamm estimate price exact amount out")
```
Comment (ethanfrey) [inline app/wasm/message_plugin.go:81]: This is shorthand for your code below... sdkerrors.Wrap(err, ...) return nil if err == nil.

But the refactor is cleaner, and allows adding events if we wish on success case.

**Review by ethanfrey — COMMENTED** [review]
Nice coverage of lots of error cases

File: app/wasm/test/messages_test.go (hunk 2 above)
Comment (maurolacy) [reply app/wasm/test/messages_test.go:106]: Good idea, I'll add a couple multi-hop checks here. Along with checking the different error conditions for them.

**Review by ethanfrey — APPROVED** [review]
lgtm

**Pull request merged**